src/mixins/Regions.js

```diff
@@ -86,12 +86,39 @@ const RegionsMixin = types
 
       const result = regions.filter(region => {
         if (excludeSelf && region === self) return false;
-        return region.dynamic && region.type === type && region.labelName === labelName;
+        const canBePartOfNotification = self.supportSuggestions ? self.dynamic : true;
+
+        return canBePartOfNotification
+          && region.type === type
+          && region.labelName === labelName
+          && region.results?.[0]?.to_name === self.results?.[0]?.to_name;
       });
 
       return result;
     },
 
+    // Indicates that it is not temporary region created just to display data like Textarea's one
+    // and is not a suggestion
+    get isRealRegion() {
+      return self.annotation?.areas?.has(self.id);
+    },
+
+    get shouldNotifyDrawingFinished() {
+      // extra calls on destroying will be skipped
+      // @see beforeDestroy action
+      if (!self.isRealRegion) return false;
+      if (self.annotation.isSuggestionsAccepting) return false;
+      // There are two modes:
+      // If object tag support suggestions - the region should be marked as a dynamic one to make notifications
+      // If object tag doesn't support suggestions - every region works as dynamic with auto suggestions
+      const canBeReasonOfNotification = self.supportSuggestions ? self.dynamic && !self.fromSuggestion : true;
+
+      const isSmartEnabled = self.results.some(r => {
+        return r.from_name.smartEnabled;
+      });
```

Comment:
```suggestion
      const isSmartEnabled = self.results.some(r => r.from_name.smartEnabled);
```
just to make it a little shorter, its a 1 liner